### PR TITLE
Bug fix in omsadmin.sh

### PR DESCRIPTION
**Pull request by robbiezhang**
Valid the workspace id is a valid GUID
Delete the certficate if onboarding failed

@Microsoft/omsagent-devs

### CODE REVIEW

**Comment by sw47** [comment]
signed off

File: installer/scripts/omsadmin.sh [hunk 1]
```diff
@@ -427,6 +433,7 @@ onboard()
     
     if [ $error -ne 0 ]; then
         log_error "Error during the onboarding request. Check the correctness of the workspace ID and shared key or run omsadmin.sh with '-v'"
+        rm "$FILE_CRT" "$FILE_KEY" > /dev/null 2>&1 || true
```
Comment (jeffaco) [inline installer/scripts/omsadmin.sh]: What if the error is transient (server outage, internet outage)? How do we keep from deleting good keys due to a transient problem?

**Comment by jeffaco** [comment]
I'm concerned about transient outages causing potential problems here. Other than that, this looks fine.

Are transient outages a concern?

File: installer/scripts/omsadmin.sh (hunk 1 above)
Comment (robbiezhang) [inline installer/scripts/omsadmin.sh]: This is the onboard method. Although the cert/key is good, but it's not registered to the OMS serve if there is a server outage, the client should not reuse the cert/key.

**Comment by jeffaco** [comment]
Okay, signed off.

File: installer/scripts/omsadmin.sh (hunk 1 above)
Comment (robbiezhang) [inline installer/scripts/omsadmin.sh]: Please note that, it's not the renew certificate scenario, where we will backup the old certificate, and overwritten when the renew succeed.

**Pull request merged**